shared-libs/rules-engine/src/rules-state-store.js

```diff
@@ -42,6 +42,7 @@ const self = {
 
     const rulesConfigHash = hashRulesConfig(settings);
     if (state.rulesConfigHash !== rulesConfigHash || targetState.isStale(state.targetState)) {
+      state.targetState = targetState.migrateStaleState(state.targetState);
```

Comment:
It's weird to call this and immediately check `isStale` again - it would be better to split this into two `if` statements so we only call `migrateStale` when it's stale.

Reply:
thanks for the review!!!